contrib/mesos/pkg/minion/tasks/task.go

```diff
@@ -281,16 +281,49 @@ func taskRunning(t *Task) taskStateFn {
 
 	select {
 	case <-t.shouldQuit:
-		t.tryComplete(t.awaitDeath(defaultKillGracePeriod, waitCh))
+		t.tryComplete(t.awaitDeath(&defaultTimer{}, defaultKillGracePeriod, waitCh))
```

Comment:
there's no complex init to do here, so I think that a new zero-value instance is OK. adding a newTimer() func feels superfluous, especially since this is package-private. I will rename to `realTimer` though because `defaultTimer` doesn't convey enough meaning.